src/components/ModalInput/ModalInput.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Text, Button, Input, InputProps, Flex, Link } from '@pancakeswap-libs/uikit'
+import useI18n from '../../hooks/useI18n'
+
+interface ModalInputProps {
+  max: string
+  symbol: string
+  onSelectMax?: () => void
+  onChange: (e: React.FormEvent<HTMLInputElement>) => void
+  placeholder?: string
+  value: string
+  addLiquidityUrl?: string
+  withdrawal?: boolean
+}
+
+const getBoxShadow = ({ isWarning = false, theme }) => {
+  if (isWarning) {
+    return theme.shadows.warning
+  }
+
+  return theme.shadows.inset
+}
+
+const StyledTokenInput = styled.div<InputProps>`
+  display: flex;
+  flex-direction: column;
+  background-color: ${({ theme }) => theme.colors.input};
+  border-radius: 16px;
+  box-shadow: ${getBoxShadow};
+  color: ${({ theme }) => theme.colors.text};
+  padding: 8px 16px 8px 0;
+  width: 100%;
+`
+
+const StyledInput = styled(Input)`
+  box-shadow: none;
+  width: 60px;
+  margin: 0 8px;
+  padding: 0 8px;
+
+  ${({ theme }) => theme.mediaQueries.xs} {
+    width: 80px;
+  }
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    width: auto;
+  }
+`
+
+const StyledErrorMessage = styled(Text)`
+  position: absolute;
+  bottom: -22px;
+  a {
+    display: inline;
+  }
+`
+
+const ModalInput: React.FC<ModalInputProps> = ({
+  max,
+  symbol,
+  onChange,
+  onSelectMax,
+  value,
+  addLiquidityUrl,
+  withdrawal,
+}) => {
+  const TranslateString = useI18n()
+  const isBalanceZero = max === '0' || !max
```

Comment:
This condition determines a few things, including visually whether to show the red 'failure' input outline, and let the user know they have no tokens to stake.